### PR TITLE
[release-v1.10] sync upstream deps

### PR DESCRIPTION
**Pull request by matzew**
Basically, this PR runs:
```
 ./hack/update-deps.sh --upgrade --release 1.10
```

### CODE REVIEW

**Review by pierDipi — COMMENTED** [review]
You need to re-apply the vendor patches

**Comment by matzew** [comment]
> You need to re-apply the vendor patches
MAKE RELEASE=1.11 is not giving me any diff

File: vendor/knative.dev/pkg/controller/stats_reporter.go [hunk 1]
```diff
@@ -199,7 +199,7 @@ func (r *reporter) ReportReconcile(duration time.Duration, success string, key t
 		return err
 	}
 
-	// TODO skonto: fix latency histogram
```
Comment (matzew) [inline vendor/knative.dev/pkg/controller/stats_reporter.go:202]: this should not be removed ...

/hold

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**